packages/graphback-runtime-knex/src/MysqlDBKnexDataProvider.ts

```diff
@@ -0,0 +1,50 @@
+import { getDatabaseArguments } from '@graphback/core';
+import { GraphQLObjectType } from 'graphql';
+import * as Knex from 'knex';
+import { KnexDBDataProvider } from './KnexDBDataProvider';
+import { NoDataError } from './NoDataError';
+
+/**
+ * Knex.js database data provider exposing basic CRUD operations.
+ *
+ * NOTE: This class implements Postgres specific implementaion that provides more performant object creation than generic `KnexDBDataProvider`
+ * that works with the rest of the databases.
+ */
+//tslint:disable-next-line: no-any
+export class MysqlDBKnexDBDataProvider<Type = any, GraphbackContext = any> extends KnexDBDataProvider<Type, GraphbackContext>{
+
+
+    public constructor(baseType: GraphQLObjectType, db: Knex) {
+        super(baseType, db);
+    }
+
+    public async create(data: Type): Promise<Type> {
+
+        const { data: createData } = getDatabaseArguments(this.tableMap, data);
+
+        // check availability of the table
+        const result = await this.db(this.tableName).columnInfo().then();
+
+        if(Object.keys(result).length == 0){
+            //creating table
+            const tableRes:any = await this.db.schema.createTable(this.tableName, function (table) {
+                table.increments();
```

Comment:
This is the data persistence layer and the database structure should never be modified here.